Add service account token refresh to AWS Container Insights Kueue Receiver

rvasahu-amazon opened the pull request:
#### Description

This PR:

1. Updates the `KueuePrometheusScraper` to specify service account token file path via `HTTPConfig` for refreshing the token as necessary.
2. Other minor documentation and renaming changes.

#### Link to tracking issue

Tracking issues are internal.

#### Testing

Through manual testing, I have verified:

1. Metrics cannot be collected without a token present at the specified file path.
2. When a valid token is added, the scrape succeeds.
3. If/when the token becomes invalid, scrapes no longer succeeds.
4. When the token is rotated, there is no gap in metrics collection.

#### Documentation

No documentation was added.

rvasahu-amazon committed:
fix: add service account token refresh to Kueue Prometheus Scraper

change: remove ``BearerToken`` from ACI Kueue Receiver

doc: minor comment and function name updates for accuracy

chadpatel commented:
You need to add #nosec to the end of the line to fix the build

```
	svcAcctTokenPath    = "/var/run/secrets/kubernetes.io/serviceaccount/token" // #nosec
```

rvasahu-amazon committed:
build: update go.mod

build: exempt bearer token filepath from security checks (not a
hardcoded credential)
